### PR TITLE
Feature/505 add my projects page

### PR DESCRIPTION
**Pull request by macfleury-2000**
## Description

A few sentences describing the overall goals of the pull request's commits.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [X] I updated the changelog with an end-user readable description
-   [X] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Login with identity server or FHICT account
2. If the logged in user hasn't uploaded any projects, upload a project
3. Hover over your profile picture (top right corner)
4. Click on my projects
5. See (all) your project(s) appear here

https://github.com/DigitalExcellence/dex-frontend/issues/505

Closes: #505

### CODE REVIEW

File: src/app/components/auth-callback/auth-callback.component.css.map [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "version": 3,
+    "mappings": "AAAA;;;;;;;;;;;;;;;;;GAiBG",
+    "sources": [
+        "auth-callback.component.scss"
+    ],
+    "names": [],
+    "file": "auth-callback.component.css"
+}
```
Comment (niraymak) [inline src/app/components/auth-callback/auth-callback.component.css.map:9]: What does this file do? I guess it maps the .css to .scss? Why not use .scss?

File: src/app/modules/home/main/main.component.css [hunk 2]
```diff
@@ -0,0 +1 @@
+/* No CSS *//*# sourceMappingURL=main.component.css.map */
```
Comment (niraymak) [inline src/app/modules/home/main/main.component.css:1]: Is this file necessary since it's empty and there is a .scss?

File: src/app/components/auth-callback/auth-callback.component.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/*
```
Comment (niraymak) [inline src/app/components/auth-callback/auth-callback.component.css:1]: Is this file necessary since it's empty and there is a .scss?

File: src/app/modules/home/main/main.component.css.map [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (niraymak) [inline src/app/modules/home/main/main.component.css.map:1]: What does this file do? I guess it maps the .css to .scss? Why not use .scss?

File: src/app/modules/user/user-projects/user-projects.component.ts [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+  /**
+   * Boolean to determine whether the component is loading the information from the api.
+   */
+  public projectsLoading = true;
+
+  public showListView = false;
+
+  public showPaginationFooter = true;
+  public currentPage;
```
Comment (niraymak) [inline src/app/modules/user/user-projects/user-projects.component.ts:59]: missing comments

File: src/app/modules/user/user-projects/user-projects.component.ts [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+   * Boolean to determine whether the component is loading the information from the api.
+   */
+  public userProjectsLoading = true;
+
+  /**
+   * Project parameter gets updated per project detail modal
+   */
+  public currentProject: Project = null;
+
+  private modalRef: BsModalRef;
```
Comment (niraymak) [inline src/app/modules/user/user-projects/user-projects.component.ts:76]: missing comments

**Review by niraymak — CHANGES_REQUESTED** [review]
Thanks for your effort! :) I had a few remarks, can you please check them out?

When I loaded the "my projects" (logged in as a user with no uploaded projects) a infinite loading icon showed up.

Something seems to go wrong in mobile view, I guess because of the userinfo : 
image

File: src/app/modules/home/main/main.component.css (hunk 2 above)
Comment (macfleury-2000) [reply src/app/modules/home/main/main.component.css:1]: I think this is from a scss plugin in my IDE. I removed it

File: src/app/components/auth-callback/auth-callback.component.css.map (hunk 1 above)
Comment (macfleury-2000) [reply src/app/components/auth-callback/auth-callback.component.css.map:9]: I think it came from a SCSS plugin in my IDE. I removed it

File: src/app/modules/home/main/main.component.css (hunk 2 above)
Comment (macfleury-2000) [reply src/app/modules/home/main/main.component.css:1]: I think it came from a SCSS plugin in my IDE. I removed it

File: src/app/components/auth-callback/auth-callback.component.css (hunk 3 above)
Comment (macfleury-2000) [reply src/app/components/auth-callback/auth-callback.component.css:1]: I think it came from a SCSS plugin in my IDE. I removed it

File: src/app/modules/user/user-projects/user-projects.component.ts (hunk 5 above)
Comment (macfleury-2000) [reply src/app/modules/user/user-projects/user-projects.component.ts:59]: Added comments!

File: src/app/modules/user/user-projects/user-projects.component.ts (hunk 6 above)
Comment (macfleury-2000) [reply src/app/modules/user/user-projects/user-projects.component.ts:76]: Added comments!

File: src/app/modules/home/main/main.component.css.map (hunk 4 above)
Comment (macfleury-2000) [reply src/app/modules/home/main/main.component.css.map:1]: I think it came from a SCSS plugin in my IDE. I removed it

**Review by RubenFricke — CHANGES_REQUESTED** [review]
Good job Fleur! Looks really good and I think this will be a great feature to have! I have 2 small remarks.

In my opinion, it's not expected that the 'my projects'-option is automatically selected. 
image

I also noticed that when I opened a modal and hit the browser's back button, I was sent back a page instead of just closing the modal just like the project overview page.

Overall it looked really good, thanks for your contribution.

**Comment by niraymak** [comment]
> Good job Fleur! Looks really good and I think this will be a great feature to have! I have 2 small remarks.
> 
> In my opinion, it's not expected that the 'my projects'-option is automatically selected.
> image
> 
> I also noticed that when I opened a modal and hit the browser's back button, I was sent back a page instead of just closing the modal just like the project overview page.
> 
> Overall it looked really good, thanks for your contribution.



> Good job Fleur! Looks really good and I think this will be a great feature to have! I have 2 small remarks.
> 
> In my opinion, it's not expected that the 'my projects'-option is automatically selected.
> image
> 
> I also noticed that when I opened a modal and hit the browser's back button, I was sent back a page instead of just closing the modal just like the project overview page.
> 
> Overall it looked really good, thanks for your contribution.

The my projects option is only selected if you are on the page for me.
Tried to refactor the behaviour when pressing "back". Didn't get it working unfortunately. On the home page the modals have the same behaviour.

**Review by niraymak — APPROVED** [review]

**Review by RubenFricke — APPROVED** [review]
